### ISSUE TITLE
Page break position changes depending on display device pixel ratio

### ISSUE THREAD

**Issue by MurakamiShinyu** [vivliostyle/vivliostyle.js#1076]
## Test case 1

HTML: [depends-on-devicePixelRatio-border.html](https://gist.githack.com/MurakamiShinyu/58317731405b33a559b642dd214a82c6/raw/depends-on-devicePixelRatio-border.html)
(Gist: https://gist.github.com/MurakamiShinyu/58317731405b33a559b642dd214a82c6)

Test with Vivliostyle Viewer (v2.20.0)
https://vivliostyle.github.io/viewer/v2.20.0/#src=https://gist.githack.com/MurakamiShinyu/58317731405b33a559b642dd214a82c6/raw/depends-on-devicePixelRatio-border.html

Result in "Retina" display (window.devicePixelRatio == 2)
<img width="818" alt="Screenshot 2022-12-27 at 15 01 36" src="https://user-images.githubusercontent.com/3324737/209619036-cd3f1a71-06c0-4fda-841a-ec267693daef.png">

Result in non-Retina display (window.devicePixelRatio == 1)

![Screenshot 2022-12-27 at 15 03 56](https://user-images.githubusercontent.com/3324737/209619336-115fa57f-744c-4f65-aa3f-c41dcc720803.png)

## Test case 2

HTML: [depends-on-devicePixelRatio-img.html](https://gist.githack.com/MurakamiShinyu/47ecbe2b310b295bbbcf63e367f28861/raw/depends-on-devicePixelRatio-img.html)
(Gist: https://gist.github.com/MurakamiShinyu/47ecbe2b310b295bbbcf63e367f28861)

Test with Vivliostyle Viewer (v2.20.0)
https://vivliostyle.github.io/viewer/v2.20.0/#src=https://gist.githack.com/MurakamiShinyu/47ecbe2b310b295bbbcf63e367f28861/raw/depends-on-devicePixelRatio-img.html

Result in "Retina" display (window.devicePixelRatio == 2)
<img width="818" alt="Screenshot 2022-12-27 at 15 10 28" src="https://user-images.githubusercontent.com/3324737/209619856-87357488-0935-418d-b650-ee83e066bccb.png">

Result in non-Retina display (window.devicePixelRatio == 1)

![Screenshot 2022-12-27 at 15 12 01](https://user-images.githubusercontent.com/3324737/209619998-bb108de7-8301-4d46-90f6-6bc4ffedcebe.png)

**Comment by MurakamiShinyu** [comment]
Fixed in #1077.

Test 1: https://vivliostyle-git-fix-issue1076-vivliostyle.vercel.app/#src=https://gist.githack.com/MurakamiShinyu/58317731405b33a559b642dd214a82c6/raw/depends-on-devicePixelRatio-border.html

Result in "Retina" display (window.devicePixelRatio == 2)

<img width="1413" alt="Screenshot 2022-12-27 at 18 30 30" src="https://user-images.githubusercontent.com/3324737/209645323-5812fcc6-5e40-47ee-a195-cd783d0bc17f.png">

Test 2: https://vivliostyle-git-fix-issue1076-vivliostyle.vercel.app/#src=https://gist.githack.com/MurakamiShinyu/47ecbe2b310b295bbbcf63e367f28861/raw/depends-on-devicePixelRatio-img.html

Result in "Retina" display (window.devicePixelRatio == 2)

<img width="1415" alt="Screenshot 2022-12-27 at 18 34 41" src="https://user-images.githubusercontent.com/3324737/209645977-d407b999-ffc0-4ce4-865e-0357f89a7d31.png">

Now these results in "Retina" display are same as the results in non-Retina display (window.devicePixelRatio == 1).

**Comment by MurakamiShinyu** [comment]
## Test case 3

HTML: [depends-on-devicePixelRatio-border-inline.html](https://gist.githack.com/MurakamiShinyu/74608c66496c0984187f5246080ab181/raw/depends-on-devicePixelRatio-border-inline.html)
(Gist: https://gist.github.com/MurakamiShinyu/74608c66496c0984187f5246080ab181)

Test with Vivliostyle Viewer:
- Before this fix (v2.20.0): https://vivliostyle.github.io/viewer/v2.20.0/#src=https://gist.githack.com/MurakamiShinyu/74608c66496c0984187f5246080ab181/raw/depends-on-devicePixelRatio-border-inline.html
- After this fix: https://vivliostyle-git-fix-issue1076-vivliostyle.vercel.app/#src=https://gist.githack.com/MurakamiShinyu/74608c66496c0984187f5246080ab181/raw/depends-on-devicePixelRatio-border-inline.html

**Comment by MurakamiShinyu** [comment]
I found this problem still exist when testing [Test case 3](https://gist.githack.com/MurakamiShinyu/74608c66496c0984187f5246080ab181/raw/depends-on-devicePixelRatio-border-inline.html) in the [previous comment](https://github.com/vivliostyle/vivliostyle.js/issues/1076#issuecomment-1365981768) with Vivliostyle CLI's build command.

First, I tested the current version (cli: 6.0.0; core: 2.20.0) with the following command on Mac terminal in Retina display and also in non-Retina display.

```
vivliostyle build https://gist.githack.com/MurakamiShinyu/74608c66496c0984187f5246080ab181/raw/depends-on-devicePixelRatio-border-inline.html
```

Result PDFs:
- output in Retina display: [output-cli-core2.20.0-retina.pdf](https://github.com/vivliostyle/vivliostyle.js/files/10334056/output-cli-core2.20.0-retina.pdf)
- output in non-Retina display: [output-cli-core2.20.0-nonretina.pdf](https://github.com/vivliostyle/vivliostyle.js/files/10334058/output-cli-core2.20.0-nonretina.pdf)

Screenshot of the PDF output in Retina display:
<img width="1638" alt="Screenshot 2023-01-03 at 12 33 21" src="https://user-images.githubusercontent.com/3324737/210296978-2e5ab9eb-3b6f-4a00-85b0-22bcf587b4de.png">

Screenshot of the PDF output in non-Retina display:
<img width="1638" alt="Screenshot 2023-01-03 at 12 33 11" src="https://user-images.githubusercontent.com/3324737/210296994-9a264a78-cc78-49a0-b1e9-b423e91e4303.png">

The result on Retina display has layout problem. This is because devicePixelRatio in Retina display is 2, so the minimum border width is 0.5px in layout processing, but on Chromium's PDF output, the minimum border width is always 1px, and that inconsistency causes the layout problem in PDF output.

I thought this problem was resolved in PR #1077 (and the next PR #1079 extends that approach to enable thinner border width).

However, I tested Vivliostyle CLI with the Viewer fixed in PR #1077 and found the problem was not solved.

```
vivliostyle build --viewer https://vivliostyle-git-fix-issue1076-vivliostyle.vercel.app/ https://gist.githack.com/MurakamiShinyu/74608c66496c0984187f5246080ab181/raw/depends-on-devicePixelRatio-border-inline.html
```

Result PDFs:
- output in Retina display: [output-cli-core.pr1077-retina.pdf](https://github.com/vivliostyle/vivliostyle.js/files/10334146/output-cli-core.pr1077-retina.pdf)
- output in non-Retina display: [output-cli-core.pr1077-nonretina.pdf](https://github.com/vivliostyle/vivliostyle.js/files/10334150/output-cli-core.pr1077-nonretina.pdf)

The results are same as the previous results. The output in Retina has the same layout problem.

------

I investigated this problem and found that there is a problem in Chromiums `--headless=chrome` mode used in Vivliostyle CLI. In that mode, the `window.diviceRatio` value becomes 1, however, the minimum border width in layout processing is 0.5px, and that inconsistency causes the problem.

I made a test HTML file that contains JavaScript that checks the devicePixelRatio and minimum border width:
- [checkDevicePixelRatio.html](https://gist.githack.com/MurakamiShinyu/265968c3bab8146028e6e345b9b7bb95/raw/checkDevicePixelRatio.html)

```
vivliostyle build https://gist.githack.com/MurakamiShinyu/265968c3bab8146028e6e345b9b7bb95/raw/checkDevicePixelRatio.html
```

Result PDFs:
- output in Retina display: [checkDevicePixelRatio-result-in-retina.pdf](https://github.com/vivliostyle/vivliostyle.js/files/10334179/checkDevicePixelRatio-result-in-retina.pdf)
- output in non-Retina display: [checkDevicePixelRatio-result-in-nonretina.pdf](https://github.com/vivliostyle/vivliostyle.js/files/10334180/checkDevicePixelRatio-result-in-nonretina.pdf)

Screenshot of the PDF output in Retina display:
<img width="758" alt="Screenshot 2023-01-03 at 13 54 58" src="https://user-images.githubusercontent.com/3324737/210302243-406eed16-b2c9-4ba2-824d-99857ec376ba.png">

Screenshot of the PDF output in non-Retina display:
<img width="794" alt="Screenshot 2023-01-03 at 14 00 25" src="https://user-images.githubusercontent.com/3324737/210302419-051a4816-2cc5-4a95-9f02-3db251cbc545.png">

Also I tested it with the CLI preview command and confirmed that the results in both Retina and non-Retina displays have no problem.

```
vivliostyle preview https://gist.githack.com/MurakamiShinyu/265968c3bab8146028e6e345b9b7bb95/raw/checkDevicePixelRatio.html
```

Screenshot of preview output in Retina display:
<img width="963" alt="Screenshot 2023-01-03 at 13 45 40" src="https://user-images.githubusercontent.com/3324737/210301382-074cc255-b667-4ae9-be11-dfb423be3fa2.png">

Screenshot of preview output in non-Retina display:
![Screenshot 2023-01-03 at 13 38 52](https://user-images.githubusercontent.com/3324737/210301400-fc04183d-fa6c-4eca-9b89-c0f85ea40fcd.png)
